snippets/Google.Storage.V1.Snippets/StorageClientSnippets.cs

```diff
@@ -12,33 +12,212 @@
 
         public StorageClientSnippets(StorageSnippetFixture fixture)
         {
-            this._fixture = fixture;
+            this.fixture = fixture;
         }
 
         [Fact]
-        public void ListObjects()
+        void ListBuckets()
         {
-            var bucketId = _fixture.BucketId;
+            var projectId = fixture.ProjectId;
+
+            // <ListBuckets>
+            var client = StorageClient.Create();
+
+            // List all buckets associated with a project
+            var buckets = client.ListBuckets(projectId);
+            // </ListBuckets>
+        }
+
+        [Fact]
+        void CreateBucket()
+        {
+            var projectId = fixture.ProjectId;
+
+            // <Buckets.Insert>
+            var client = StorageClient.Create();
+
+            // GCS bucket names must be globally unique
+            var bucketName = Guid.NewGuid().ToString();
+
+            // Bucket defined in Google.Apis.Storage.v1.Data namespace
+            var bucket = client.Service.Buckets.Insert(new Bucket { Name = bucketName }, projectId).Execute();
+            // </Buckets.Insert>
+
+            Assert.Equal(bucketName, bucket.Name);
+            Assert.True(!string.IsNullOrWhiteSpace(bucket.Id));
+            client.Service.Buckets.Delete(bucketName).Execute();
+        }
+
+        [Fact]
+        void ListObjects()
+        {
+            var bucketName = fixture.BucketName;
 
             // <ListObjects>
             var client = StorageClient.Create();
-            // List only objects with a name starting with "Hello"
-            var objects = client.ListObjects(bucketId, "Hello");
+
+            // List only objects with a name starting with "greet"
+            var objects = client.ListObjects(bucketName, "greet");
             // </ListObjects>
+
             var names = objects.Select(obj => obj.Name).ToList();
-            Assert.Contains(_fixture.HelloWorldName, names);
+            Assert.Contains(fixture.HelloWorldName, names);
         }
+
+        [Fact]
+        void DownloadFile()
+        {
+            var bucketName = fixture.BucketName;
+            var projectId = fixture.ProjectId;
+
+            // <DownloadObject>
+            var client = StorageClient.Create();
+            var source = "greetings/hello.txt";
+            var destination = @".\hello.txt";
+
+            using (var stream = File.Create(destination))
+            {
+                // IDownloadProgress defined in Google.Apis.Download namespace
+                var progress = new Progress<IDownloadProgress>(
+                    p => Console.WriteLine($"bytes: {p.BytesDownloaded}, status: {p.Status}")
+                );
+
+                // Download source object from bucket to local file system
+                client.DownloadObject(bucketName, source, stream, null, progress);
+            }
+            // </DownloadObject>
+
+            // want to show the source path in the snippet to give an idea
```

Comment:
Ack. Will try to think of a cunning way of avoiding this as a problem if we run into it a lot. For the moment, this seems like a good workaround.